algebird-benchmark/src/main/scala/com/twitter/algebird/benchmark/QTreeBenchmark.scala

```diff
@@ -50,12 +61,33 @@ class QTreeBenchmark {
   import QTreeBenchmark._
 
   @Benchmark
-  def timeSumOption(state: QTreeState) = {
-    state.qtree.sumOption(state.inputData)
+  def timeSumOptionUnit(state: QTreeState) = {
+    state.qtreeUnit.sumOption(state.inputDataUnit)
+  }
+
+  @Benchmark
+  def timeSumOptionLong(state: QTreeState) = {
+    state.qtreeLong.sumOption(state.inputDataLong)
+  }
+
+  @Benchmark
+  def timeSumOptionDouble(state: QTreeState) = {
+    state.qtreeDouble.sumOption(state.inputDataDouble)
   }
 
   @Benchmark
-  def timeOldSumOption(state: QTreeState) = {
-    state.oldqtree.sumOption(state.inputData)
+  def timePlusUnit(state: QTreeState): QTree[Unit] = {
+    state.inputDataUnit.tail.foldLeft(state.inputDataUnit.head)(state.qtreeUnit.plus)
```

Comment:
maybe just use `inputDataUnit.reduce(state.qtreeUnit.plus)` `.tail` could be O(N) on a bad Seq (some require copying, I believe).